### PR TITLE
Enable CodeQL runs on PRs and weekly

### LINKED ISSUE #1324: enable CodeQL on all PRs, not just code on main
Tracking issue for:
- [ ] https://github.com/sigstore/cosign/security/code-scanning/67

### PR DESCRIPTION
**Pull request by bobcallaway**
Fixes #1324

Signed-off-by: Bob Callaway <bob.callaway@gmail.com>

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by dlorenc** [comment]
So we need it on PR runs as well as the cron? It takes an incredibly long time and almost never fails so I've been trying to leave it for post-submit only.

I think we've only ever had one failure here and it was a false positive.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -34,10 +38,31 @@ jobs:
...
+          ~/go/pkg/mod
+          ~/.cache/go-build
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Setup Go Version
+      uses: actions/setup-go@424fc82d43fa5a37540bae62709ddcc23d9520d4 #v2.1.5
+      with:
+        go-version: '1.16.x'
```
Comment (cpanato) [inline .github/workflows/codeql-analysis.yml:64]: why 1.16 and not 1.17?

**Comment by cpanato** [comment]
> So we need it on PR runs as well as the cron? It takes an incredibly long time and almost never fails so I've been trying to leave it for post-submit only.
> 
> I think we've only ever had one failure here and it was a false positive.

i agree, a post-job is good and this does not hurt much us

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/codeql-analysis.yml:64]: @bobcallaway we are using go1.18, can you update to that?

**Review by cpanato — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/codeql-analysis.yml:64]: now maybe 1.19 :)

**Comment by hectorj2f** [comment]
I've seen PR checks to slow down a lot, we need to measure how much time, it will take. Otherwise I'll suggest to run them post push.

**Pull request closed without merging**